File: verify.js
#!/usr/bin/env node

/**
 * Game Rental System - Setup Verification
 * Run this after setup to verify everything is working
 */

import fetch from 'node-fetch';
import fs from 'fs';
import path from 'path';

const API_URL = 'http://localhost:3000/api';
const checks = [];

function log(type, message) {
  const icons = {
    success: '✅',
    error: '❌',
    info: 'ℹ️ ',
    warn: '⚠️ '
  };
  console.log(`${icons[type]} ${message}`);
}

async function checkBackendRunning() {
  try {
    const response = await fetch(`${API_URL.replace('/api', '')}/health`);
    if (response.ok) {
      log('success', 'Backend is running on http://localhost:3000');
      checks.push({ name: 'Backend Running', status: true });
      return true;
    }
  } catch (error) {
    log('error', 'Backend is not running. Start it with: cd backend && npm start');
    checks.push({ name: 'Backend Running', status: false });
    return false;
  }
}

async function checkDatabase() {
  try {
    const dbPath = path.join(process.cwd(), 'backend', 'database.sqlite');
    if (fs.existsSync(dbPath)) {
      log('success', 'Database file exists');
      checks.push({ name: 'Database File', status: true });
      return true;
    } else {
      log('warn', 'Database file not found. It will be created on first backend start.');
      checks.push({ name: 'Database File', status: false });
      return false;
    }
  } catch (error) {
    log('error', `Database check failed: ${error.message}`);
    checks.push({ name: 'Database File', status: false });
    return false;
  }
}

async function checkAuth() {
  try {
    // Try login
    const response = await fetch(`${API_URL}/auth/login`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        email: 'admin@game.com',
        password: 'password'
      })
    });

    if (response.ok) {
      const data = await response.json();
      if (data.token) {
        log('success', 'Authentication working (test login successful)');
        checks.push({ name: 'Authentication', status: true });
        return data.token;
      }
    }

    log('error', 'Authentication check failed. Make sure database is seeded.');
    checks.push({ name: 'Authentication', status: false });
    return null;
  } catch (error) {
    log('error', `Auth check failed: ${error.message}`);
    checks.push({ name: 'Authentication', status: false });
    return null;
  }
}

async function checkGamesAPI(token) {
  try {
    const response = await fetch(`${API_URL}/games`, {
      headers: { 'Authorization': `Bearer ${token}` }
    });

    if (response.ok) {
      const games = await response.json();
      log('success', `Games API working (${games.length} games in database)`);
      checks.push({ name: 'Games API', status: true });
      return true;
    }
  } catch (error) {
    log('error', `Games API check failed: ${error.message}`);
    checks.push({ name: 'Games API', status: false });
    return false;
  }
}

async function checkEnvironment() {
  try {
    const envPath = path.join(process.cwd(), '.env');
    if (fs.existsSync(envPath)) {
      const content = fs.readFileSync(envPath, 'utf8');
      if (content.includes('PORT=') && content.includes('JWT_SECRET=')) {
        log('success', '.env file is configured');
        checks.push({ name: '.env Configuration', status: true });
        return true;
      }
    }
    log('warn', '.env file missing or incomplete');
    checks.push({ name: '.env Configuration', status: false });
    return false;
  } catch (error) {
    log('error', `.env check failed: ${error.message}`);
    checks.push({ name: '.env Configuration', status: false });
    return false;
  }
}

async function runVerification() {
  console.clear();
  console.log('🎮 Game Rental System - Setup Verification\n');
  console.log('═'.repeat(50));
  console.log('');

  // Check files
  console.log('📁 Checking files...');
  await checkEnvironment();
  await checkDatabase();
  console.log('');

  // Check backend
  console.log('🔌 Checking backend...');
  const backendRunning = await checkBackendRunning();

  if (!backendRunning) {
    console.log('');
    console.log('❌ Backend must be running to continue checks.');
    console.log('   Start it with: cd backend && npm start');
    console.log('');
    printSummary();
    return;
  }

  console.log('');

  // Check API
  console.log('🔐 Checking API...');
  const token = await checkAuth();

  if (!token) {
    console.log('');
    console.log('⚠️  Could not authenticate. Make sure:');
    console.log('   1. Backend has initialized the database');
    console.log('   2. seed.sql has been executed');
    console.log('   3. Test credentials exist (admin@game.com / password)');
    console.log('');
    printSummary();
    return;
  }

  console.log('');
  console.log('📊 Checking data...');
  await checkGamesAPI(token);

  console.log('');
  printSummary();

  // Success message
  console.log('');
  console.log('═'.repeat(50));
  const allPassed = checks.every(c => c.status);
  if (allPassed) {
    console.log('');
    console.log('✅ All checks passed!');
    console.log('');
    console.log('🚀 Next steps:');
    console.log('   1. Serve frontend: python -m http.server 5500');
    console.log('   2. Open browser: http://localhost:5500');
    console.log('   3. Login with: admin@game.com / password');
    console.log('');
  }
}

function printSummary() {
  console.log('═'.repeat(50));
  console.log('');
  console.log('📋 Summary:');
  checks.forEach(check => {
    const status = check.status ? '✅' : '❌';
    console.log(`  ${status} ${check.name}`);
  });
  console.log('');
}

// Run verification
runVerification().catch(error => {
  log('error', `Verification failed: ${error.message}`);
  process.exit(1);
});

File: backend/verify.py
#!/usr/bin/env python3
"""
Verification script for Game Rental System setup
Tests if backend is working correctly
"""
import subprocess
import time
import requests
import sys
import json
from pathlib import Path

def check_files():
    """Check if required files exist"""
    print("📁 Checking files...")
    files = [
        'run.py',
        'app.py',
        'database.py',
        'utils.py',
        '.env',
        'requirements.txt',
        'db/schema.sql',
        'db/seed.sql',
        'services/auth_service.py',
        'services/games_service.py',
        'services/rentals_service.py',
        'routes/auth_routes.py',
        'routes/games_routes.py',
        'routes/rentals_routes.py'
    ]
    
    missing = []
    for f in files:
        if not Path(f).exists():
            missing.append(f)
    
    if missing:
        print(f"  ❌ Missing files: {', '.join(missing)}")
        return False
    
    print("  ✅ All files present")
    return True

def check_dependencies():
    """Check if dependencies are installed"""
    print("📦 Checking dependencies...")
    try:
        import flask
        import flask_cors
        import flask_jwt_extended
        import dotenv
        import bcrypt
        print("  ✅ All dependencies installed")
        return True
    except ImportError as e:
        print(f"  ❌ Missing dependency: {e}")
        print("     Run: pip install -r requirements.txt")
        return False

def check_database():
    """Check if database exists"""
    print("💾 Checking database...")
    db_path = Path('database.sqlite')
    if db_path.exists():
        print("  ✅ Database file exists")
        return True
    else:
        print("  ℹ️  Database will be created on first run")
        return True

def check_backend(url='http://localhost:3000'):
    """Check if backend is running"""
    print("🔌 Checking backend...")
    try:
        response = requests.get(f'{url}/health', timeout=2)
        if response.status_code == 200:
            print("  ✅ Backend is running")
            return True
    except:
        print(f"  ❌ Backend not running at {url}")
        print("     Start it with: python run.py")
        return False

def check_auth(url='http://localhost:3000'):
    """Test authentication"""
    print("🔐 Testing authentication...")
    try:
        # Try login with test credentials
        response = requests.post(
            f'{url}/api/auth/login',
            json={
                'email': 'admin@game.com',
                'password': 'password'
            },
            timeout=5
        )
        
        if response.status_code == 200:
            data = response.json()
            if 'token' in data:
                print("  ✅ Authentication working")
                return True
        
        print("  ❌ Authentication failed")
        return False
    except Exception as e:
        print(f"  ❌ Error: {e}")
        return False

def check_api(url='http://localhost:3000'):
    """Test API endpoints"""
    print("📡 Testing API...")
    try:
        # Test games endpoint
        response = requests.get(f'{url}/api/games', timeout=5)
        if response.status_code == 200:
            games = response.json()
            print(f"  ✅ Games API working ({len(games)} games)")
            return True
        
        print("  ❌ Games API failed")
        return False
    except Exception as e:
        print(f"  ❌ Error: {e}")
        return False

def main():
    """Run all checks"""
    print("\n" + "="*50)
    print("Game Rental System - Verification Script")
    print("="*50 + "\n")
    
    results = {
        'Files': check_files(),
        'Dependencies': check_dependencies(),
        'Database': check_database(),
    }
    
    print("\n" + "-"*50)
    print("Make sure backend is running for next checks...")
    print("-"*50 + "\n")
    
    results['Backend Running'] = check_backend()
    
    if results['Backend Running']:
        results['Authentication'] = check_auth()
        results['API'] = check_api()
    
    print("\n" + "="*50)
    print("Summary:")
    print("="*50)
    
    for check, result in results.items():
        status = "✅" if result else "❌"
        print(f"{status} {check}")
    
    all_passed = all(results.values())
    
    if all_passed:
        print("\n✅ Everything looks good! System is ready to use.")
    else:
        print("\n❌ Some checks failed. See above for details.")
    
    return 0 if all_passed else 1

if __name__ == '__main__':
    sys.exit(main())
